### PR TITLE
exclude past time in offers' availabilities

### PR DESCRIPTION
**Pull request by DanNiESh**

### CODE REVIEW

File: esi_leap/objects/offer.py [hunk 1]
```diff
@@ -63,9 +65,14 @@ def get_availabilities(self):
             return []
 
         conflicts = self.dbapi.offer_get_conflict_times(self)
-
+        now = datetime.datetime.now()
```
Comment (tzumainn) [inline esi_leap/objects/offer.py:68]: I think after this, I'd have

```
 start_time = self.start_time if self.start_time >= now else now
```
Then I'd use start_time in various places below.

File: esi_leap/objects/offer.py [hunk 2]
```diff
@@ -63,9 +65,14 @@ def get_availabilities(self):
...
         if conflicts:
-            a = [self.start_time, conflicts[0][0]]
+            if now <= self.start_time:
+                a = [self.start_time, conflicts[0][0]]
+            elif now < conflicts[0][0]:
+                a = [now, conflicts[0][0]]
+            else:
+                a = []
```
Comment (tzumainn) [inline esi_leap/objects/offer.py:75]: Not your fault, but I'm having trouble understanding the logic in this entire function. Could you add some comments explaining?

File: esi_leap/tests/objects/test_offer.py [hunk 3]
```diff
@@ -92,9 +92,14 @@ def test_get_all(self, mock_offer_get_all):
         self.assertEqual(self.context, offers[0]._context)
 
     @mock.patch('esi_leap.db.sqlalchemy.api.offer_get_conflict_times')
-    def test_get_availabilities(self, mock_offer_get_conflict_times):
+    @mock.patch('esi_leap.objects.offer.datetime')
+    def test_get_availabilities(self, mock_datetime,
```
Comment (tzumainn) [inline esi_leap/tests/objects/test_offer.py:96]: I think we may want to split up this test into several smaller tests.

File: esi_leap/objects/offer.py [hunk 4]
```diff
@@ -63,27 +65,49 @@ def get_availabilities(self):
...
-                    i += 1
-
-            a = [[a[j], a[j + 1]] for j in range(0, len(a) - 1, 2)]
+                    a = []
+                    for i in range(len(conflicts) - 1):
+                        # Find the conflict timeframe that started
+                        # in the past and will end in the future.
+                        # add all times after this
+                        if (conflicts[i][0] <= now and conflicts[i][1] > now) \
+                            or conflicts[i][0] > now:
```
[the remark this answers is not in the document]
Comment (DanNiESh) [reply esi_leap/objects/offer.py:92]: Sure!

File: esi_leap/objects/offer.py [hunk 5]
```diff
@@ -63,27 +65,49 @@ def get_availabilities(self):
...
+
+                i = 0
+                while i < len(a) - 1:
+                    if a[i] == a[i + 1]:
+                        a.pop(i)
+                        a.pop(i)
+                    else:
+                        i += 1
+
+                a = [[a[j], a[j + 1]] for j in range(0, len(a) - 1, 2)]
```
[the remark this answers is not in the document]
Comment (DanNiESh) [reply esi_leap/objects/offer.py]: Fixed.

**Review by tzumainn — APPROVED** [review]
Looks good - thanks!

**Pull request merged**